hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -327,9 +307,16 @@ public Partitioner getPartitioner(DataSet<LocatedRow> dataSet, ExecRow template,
                 return handleExceptionInCaseOfEmptySet(e,location);
             }
             table = processExternalDataset(table,baseColumnMap,qualifiers,probeValue);
-            return new SparkDataSet(table
-                    .rdd().toJavaRDD()
-                    .map(new RowToLocatedRowFunction(context,execRow)));
+
+            if (useSample) {
+                return new SparkDataSet(table
+                        .rdd().toJavaRDD()
+                        .map(new RowToLocatedRowFunction(context, execRow)).sample(false, sampleFraction));
```

Comment:
@dgomezferro I've updated the PR to move sample() before map(). Same change is made to the PR for 2.5 too.